### PR TITLE
Remove ActionRequestDetails

### PR DESCRIPTION
**Pull request by jamesagnew**
This PR removes ActionRequestDetails - This class has been deprecated since HAPI FHIR 4.0.0, so it makes sense to drop it in HAPI FHIR 6.2.0.

This fixes an O(n^2) issue in SearchNarrowingInterceptor, where batch submissions containing a large number of conditional create/update operations degrade performance significantly as they get larger.

It is potentially a breaking change for anyone depending on the undocumented behaviour of the `SERVER_INCOMING_REQUEST_PRE_HANDLED` pointcut being called from within JPA operations.

### CODE REVIEW

**Review by tadgh — APPROVED** [review]
Approved pending resolution of comments

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/IServerInterceptor.java [hunk 1]
```diff
@@ -132,7 +132,7 @@ boolean handleException(RequestDetails theRequestDetails, BaseServerResponseExce
 	 *                            server, e.g. the FHIR operation type and the parsed resource body (if any).
 	 */
 	@Hook(Pointcut.SERVER_INCOMING_REQUEST_PRE_HANDLED)
-	void incomingRequestPreHandled(RestOperationTypeEnum theOperation, ActionRequestDetails theProcessedRequest);
+	void incomingRequestPreHandled(RestOperationTypeEnum theOperation, RequestDetails theProcessedRequest);
```
Comment (oliviayou) [inline hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/IServerInterceptor.java:135]: Suggestion (non-blocking): Update the Javadoc as ActionRequestDetails has been removed

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/util/UrlUtil.java [hunk 2]
```diff
@@ -62,43 +62,21 @@ public class UrlUtil {
...
-			myVersionId = theVersionId;
-		}
+	/**
+	 * Cleans up a value that will be serialized as an HTTP header. This method:
+	 *
+	 * - Strips any newline characters
+	 *
+	 * @since 6.0.0
+	 */
+	public static String sanitizeHeaderValue(String thePollLocation) {
```
[the remark this answers is not in the document]
Comment (jamesagnew) [reply hapi-fhir-base/src/main/java/ca/uhn/fhir/util/UrlUtil.java:78]: Fixed.

And good point about \r.. Technically that's not a valid header character and shouldn't be there (HTTP spec calls for \n only) but this method is all about sanitizing so it should definitely catch that.

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3527-drop-actionrequestdetails.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+type: remove
+issue: 3527
+title: "The `ActionRequestDetails` class has been dropped (it has been deprecated 
+   since HAPI FHIR 4.0.0). This class was used as a parameter to the 
+   `SERVER_INCOMING_REQUEST_PRE_HANDLED` interceptor pointcut, but can be
+   replaced in any existing client code with `RequestDetails`. This change
+   also removes an undocumented behaviour where the JPA server internally
+   invoked the `SERVER_INCOMING_REQUEST_PRE_HANDLED` a second time from
+   within various processing methods. This behaviour caused performance
+   problems for some interceptors (e.g. `SearchNarrowingInteerceptor`) and
```
[the remark this answers is not in the document]
Comment (jamesagnew) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3527-drop-actionrequestdetails.yaml]: Done

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/method/PageMethodBinding.java [hunk 4]
```diff
@@ -153,7 +150,7 @@ private IBaseResource handlePagingRequest(IRestfulServer<?> theServer, RequestDe
 		}
 
 		String linkSelfBase = theRequest.getFhirServerBase(); // myServerAddressStrategy.determineServerBase(getServletContext(),
-																				// theRequest.getServletRequest());
+		// theRequest.getServletRequest());
```
[the remark this answers is not in the document]
Comment (jamesagnew) [reply hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/method/PageMethodBinding.java]: Done

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/IServerInterceptor.java (hunk 1 above)
Comment (jamesagnew) [reply hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/IServerInterceptor.java:134]: Done

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3527-drop-actionrequestdetails.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+   no longer offers any benefit so it is being removed."
```
[the remark this answers is not in the document]
Comment (jamesagnew) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3527-drop-actionrequestdetails.yaml]: Applied

**Pull request merged**